frontend-react/src/utils/CustomRenderUtils.tsx

```diff
@@ -1,174 +1,100 @@
-);
-const FeatureFlagWrapper: FC = ({ children }) => {
-    return (
-        <BaseWrapper>
-            <FeatureFlagProvider>{children}</FeatureFlagProvider>
-        </BaseWrapper>
-    );
-};
-
-const AppWrapper = ({ children }: PropsWithChildren<{}>) => {
-    return (
-        <BaseWrapper>
-            <RouterWrapper>
-                <SessionProvider>
-                    <QueryClientProvider client={getTestQueryClient()}>
-                        <AuthorizedFetchProvider>
-                            <FeatureFlagProvider>
-                                {children}
-                            </FeatureFlagProvider>
-                        </AuthorizedFetchProvider>
-                    </QueryClientProvider>
-                </SessionProvider>
-            </RouterWrapper>
-        </BaseWrapper>
-    );
+    };
 };
 
-export const renderWithBase = (
-    ui: ReactElement,
-    options?: Omit<RenderOptions, "wrapper">
-) =>
-    render(ui, {
-        wrapper: BaseWrapper,
-        ...options,
-    });
-
-const renderWithRouter = (
-    ui: ReactElement,
-    options?: Omit<RenderOptions, "wrapper">
-) => render(ui, { wrapper: RouterWrapper, ...options });
-
-const renderWithSession = (
-    ui: ReactElement,
-    options?: Omit<RenderOptions, "wrapper">
-) =>
-    render(ui, {
-        wrapper: SessionWrapper,
-        ...options,
-    });
+interface RenderAppOptions extends RenderOptions, AppWrapperOptions {}
 
-// render an element with parent wrapper other than a div
-// used to silence testing library errors when wrapping elements
-// that should not be children of divs (such as <td> etc)
-export const renderWithCustomWrapper = (
+export const renderApp = (
     ui: ReactElement,
-    wrapperType: string
+    {
+        initialRouteEntries,
+        restHookFixtures,
+        ...options
+    }: Omit<RenderAppOptions, "wrapper"> = {}
 ) => {
     return render(ui, {
-        container: document.body.appendChild(
-            document.createElement(wrapperType)
-        ),
-    });
-};
-
-// render wrapped with BrowserRouter, SessionProvider, and QueryClientProvider
-export const renderWithFullAppContext = (
-    ui: ReactElement,
-    oktaHook?: OktaHook,
-    options?: Omit<RenderOptions, "wrapper">
-) => {
-    return render(ui, {
-        wrapper: AppWrapper,
+        wrapper: AppWrapper({ initialRouteEntries, restHookFixtures }),
         ...options,
     });
 };
 
-// for testing components that need access to react-query
-export const renderWithQueryProvider = (
-    ui: ReactElement,
-    options?: Omit<RenderOptions, "wrapper">
-) =>
-    render(ui, {
-        wrapper: QueryWrapper(),
-        ...options,
-    });
-
-// for testing components that need access to feature flags
-export const renderWithFeatureFlags = (
-    ui: ReactElement,
-    options?: Omit<RenderOptions, "wrapper">
-) =>
-    render(ui, {
-        wrapper: FeatureFlagWrapper,
-        ...options,
-    });
-
 export * from "@testing-library/react";
```

Comment:
Since we're funneling all exports from RTL, we may want to consider exporting `renderApp` as the default `render` (and eventually a default `renderHook`) and just import everything from CustomRenderUtils in our tests.  It's a suggested pattern in the official RTL docs [here](https://testing-library.com/docs/react-testing-library/setup/#custom-render), and it's something I've personally implemented at previous workplaces to pretty good effect.  It's especially useful if we need to extend the return value with, say, a query client from react-query or something.

Not something we need to do now of course!  Just wanted to get your thoughts!

Reply:
I noticed this as a recommended pattern when researching tests. It also recommends changing the config so that the folder containing the new testing exports module be loaded at root so that only a simple fixed import statement is needed rather than a relative one (import { render } from "CustomRenderUtils" vs import { render } from "../../../CustomRenderUtils") which is why I held off on making this pattern change until it was discussed. I'm for this change, whether we go the extra step on the root import or not.